drivers/storage/portworx/portworx.go

```diff
@@ -98,8 +99,30 @@ func (p *portworx) Init(
 	return nil
 }
 
+func EnableSecurityContextForValidate(recorder record.EventRecorder, cluster *corev1.StorageCluster) error {
+	if secComp, exists := component.Get(component.SCCComponentName); exists {
+		if secComp.IsEnabled(cluster) {
+			err := secComp.Reconcile(cluster)
+			if ce, ok := err.(*component.Error); ok &&
+				ce.Code() == component.ErrCritical {
+				return fmt.Errorf("pre-flight: %s component setup failed: %v", secComp.Name(), err)
+			} else if err != nil {
+				msg := fmt.Sprintf("pre-flight: failed to setup %s. %v", secComp.Name(), err)
+				k8sutil.WarningEvent(recorder, cluster, util.FailedComponentReason, msg)
+			}
+		}
+	}
+	return nil
+}
+
 func (p *portworx) Validate(cluster *corev1.StorageCluster) error {
 
+	if installSecContext {
+		if err := EnableSecurityContextForValidate(p.recorder, cluster); err == nil {
+			installSecContext = false // if no err, done enabling
```

Comment:
There is a small gap here - if we install a Portworx cluster, then uninstall it and install another one, all without restarting the operator, the second install will not have the security context enabled.

Can we either reset this variable on uninstall, or find a way to do the check more "atomically" without relying on an in-memory state variable?

Reply:
Also ideally add a unit test for this case :)

Reply:
yeah actually I can probably use the cluster object to manage this.   Like I mentioned above this is more of small optimization,  redo'ing the call would be a no-op.   But let me see if I can use a cluster status object which is better..

Reply:
It is okay if the reconcile of the scc component gets called on every operator reconcile loop. Most of the calls go the k8s cache and don't hit the api server so they are not expensive or time consuming.

Let's rename the function to `reconcileSecurityContextForValidate` and remove the `installSecContext` variable.

Reply:
What we could do is call this before we start the pre-flight on line 165. That way once the preflight is done, we don't keep calling  this as `PreInstall` will take care of the SCCs.

Reply:
@Pure-AdamuKaapan I slacked with Piyush about calling that code in the reconcile loop and he said it as ok.. in fact they are are already doing that indirectly in another place.    However we need it here cause that other place is done after pre-flight is run.. which is actually why this PR was filed.  ```:spiral_calendar_pad:  [1:29 PM](https://purestorage.slack.com/archives/D01NTSM12HJ/p1694723387150449)
Yes, it can be called over and over. We do that in the PreInstall too, which is called by the operator every 30 seconds anyway.
  [1:30 PM](https://purestorage.slack.com/archives/D01NTSM12HJ/p1694723411687099)
Oh ok cool... thanks forgot about that .. awesome..```